libs/core/debugging/src/print.cpp

```diff
@@ -57,6 +57,10 @@ namespace hpx::debug {
             std::ostream&, std::int32_t const&, int);
         template HPX_CORE_EXPORT void print_dec(
             std::ostream&, std::int64_t const&, int);
+#ifdef __APPLE__
+        template HPX_CORE_EXPORT void print_dec(
+            std::ostream&, unsigned long const&, int);
```

Comment:
Would it be better to unconditionally add a specialization for `std::uint32_t`?

Reply:
I need more clarity on this because for x86 macOs this could be different and not sure why this isn't already handled by macOS's xcode standard SDK.
1. We wait for them to fix this
2. We keep this and it does not interfere with any of the std::uint32_t and builds
3. We do an unconditional specialization for std::uint32_t and it may override or fail to build in the future.

Reply:
Adding the specialization for `std::uint32_t` will not do any harm and should be equivalent to the one you added.